server/src/main/java/org/elasticsearch/script/field/vectors/BinaryDenseVector.java

```diff
@@ -83,6 +83,16 @@ public double l1Norm(List<Number> queryVector) {
         return l1norm;
     }
 
+    @Override
+    public int hamming(byte[] queryVector) {
+        throw new UnsupportedOperationException("hamming distance is not supported for float vectors");
+    }
```

Comment:
I was assuming that this being a `byte` query vector, it should not throw an `UOE` related to float vectors?

Reply:
Its a `byte[]` query vector against a `float` indexed vector.